sympy/integrals/prde.py

```diff
@@ -441,6 +441,52 @@ def prde_no_cancel_b_small(b, Q, n, DE):
     return f + H, A.col_join(B).col_join(C)
 
 
+def prde_cancel_primitive(b, Q, n, DE):
+    """
+    Pg, 237
+    """
+    with DecrementLevel(DE):
+        ba, bd = frac_in(b, DE.t)
+        Qyn = [frac_in(q.nth(n), DE.t, field=True) for q in Q]
+        # fn = f1n, ..., frn_n
+        # An is a matrix with coefficients in Const(k)
+        # yn = Sum(djn*fjn, (j, 1, rn))
+        fn, An = param_rischDE(ba, bd, Qyn, DE)
```

Comment:
`param_rischDE` should be called repeatedly for descending degrees n, n -1, ..., 0, and the results should probably be saved in a list like `f[n]`, `A[n]` to be combined at the end before returning. Alternatively, it might be possible to save some list space by doing a partial combination of the results after each call of `param_rischDE`.